core/unit_test/headers_self_contained/CMakeLists.txt

```diff
@@ -10,7 +10,7 @@ file(GLOB KOKKOS_CONTAINERS_HEADERS RELATIVE ${BASE_DIR}/containers/src
 file(GLOB KOKKOS_ALGORITHMS_HEADERS RELATIVE  ${BASE_DIR}/algorithms/src
      ${BASE_DIR}/algorithms/src/*.hpp)
 
-if(NOT Kokkos_ENABLE_DEPRECATED_CODE_4)
+if(Kokkos_ENABLE_DEPRECATED_CODE_4 AND Kokkos_ENABLE_DEPRECATION_WARNINGS)
```

Comment:
OK so now we hit
https://github.com/kokkos/kokkos/blob/835dbf5948f1ae8da38a4b320e2f2dd789a54c6d/containers/src/Kokkos_Vector.hpp#L37

I suppose it means we should just unconditionally exclude that test :/